test/suites/remote.sh

```diff
@@ -26,6 +26,11 @@ test_remote_url() {
     lxc_remote remote add ubuntu2 https://cloud-images.ubuntu.com:443/releases --protocol=simplestreams
     lxc_remote remote remove ubuntu1
     lxc_remote remote remove ubuntu2
+
+    # a connectivity issue returns an error
+    ! lxc_remote remote add ubuntu1 https://cloud-images.ubuntu.com:1234/releases --protocol=simplestreams
+    # a malformed url path returns an error
+    ! lxc_remote remote add ubuntu2 https://cloud-images.ubuntu.com/buildd/releases/ --protocol=simplestreams --public
```

Comment:
If the issue is the trailing "/", I would update the comment to say that trailing "/" are not permitted, rather than it being a malformed URL.

Reply:
Ive not had time to debug it much, but that line is one of the issues.

The other thing is that it broke adding valid simplestreams URLs without `--public` being specified as it then treated them as LXD image servers.